### DISCUSSION TITLE
exec_mem_limit的使用问题

### DISCUSSION THREAD

**Discussion by tryniyijiao** [apache/doris#12550] [A - General / Q&A]
当我查询一张4千万表的时候，默认exec_mem_limit的2G大小会报内存不足异常，增加大小可以解决问题，但是当表的数据量继续增大时，这里的exec_mem_limit是不是又要给的更大，有其他更好的解决方案吗? 当数据量更大的时候，exec_mem_limit参数的值也不可能设置无限大
一条很普通的sql:
select
  event_sid,
  date_format(event_time, 'yyyy-MM-dd') as eventDate,
  COUNT(distinct(event_user_id)) cnt
from
  dws_event_analysis
where
  1 = 1
  and event_time between '2022-07-15 00:00:00'
  and '2022-09-13 00:00:00'
  and event_sid = '001.0009.default.default.17140099'
  and event_org_code = '220706fUa5jbyK8999'
group by
  event_sid,
  date_format(event_time, 'yyyy-MM-dd')

**Comment by JNSimba** [comment]
`exec_mem_limit` refers to the memory of a single BE. When the data you want to query exceeds 2G on a single BE, an error will be reported.

2G memory is very small, it is recommended to set this value larger according to the machine configuration of be

**Reply by tryniyijiao** [reply]
你好，就是我不太清楚给到多大合适，官网文档也没有参考值说明，目前我的查询，给到24G还是报错，就怕后续数据量大了，这个值要设置到物理机内存上限

**Comment by dataalive** [comment]
你的sql里看起来比较耗内存的主要是`COUNT(distinct(event_user_id)) cnt` 和 `group by`这两个地方
